### PR TITLE
Expose rsyslog metrics via `node-exporter`'s textfile collector and display them in a dedicated `plutono` dashboard

### PR DESCRIPTION
**Pull request by plkokanov**
**How to categorize this PR?**

/kind enhancement

**What this PR does / why we need it**:
This PR exposes rsyslog metrics by making use of the `node-exporter`'s textfile collector. It also adds alerting rules and a `plutono` dashboard configuration where the metrics are displayed.

More concretely:
- 📈  This PR adds a script which transforms rsyslog pstats metrics into ones that can be parsed by the `node-exporter`'s textfile collector and saves them in `/var/lib/node-exporter/textfile-collector/rsyslog-pstats.prom`. 
- ℹ️ With https://github.com/gardener/gardener/pull/8721 the textfile collector of the `node-exporter` is enabled and it parses all files that match the `*.prom` glob in the `/var/lib/node-exporter/textfile-collector` directory. Any metrics that it finds in those files are exposed on its `/metrics` endpoint.
- 🧑‍🔬 A prometheus scrape config is added which reads all metrics which match the `rsyslog_.+` regex pattern from the `node-exporter`'s `/metrics` endpoint
- 📟 Alerting rules which fire when there are too many failed rsyslog relp actions or when there are zero successful rsyslog relp actions indicating a potential problem with the rsyslog service
- 📊 A dedicated dashboard in the shoot `plutono` which can be used to monitor the rsyslog service.

**Which issue(s) this PR fixes**:
Part of #2

**Special notes for your reviewer**:
Still a draft as I want to revendor to a gardener version in which the text file collector is enabled.

**Release note**:

```other operator
Metrics for the rsyslog service running on the shoot nodes are now exposed and collected according to the following:
  - The metrics are available on the `node-exporter`'s `/metrics` endpoint. 
  - The names of the new metrics match the `rsyslog_pstat_.+` regex.
  - The metrics are scraped and collected in the shoot's prometheus instance.
  - A dedicated plutono dashboard is added which displays the rsyslog metrics.
```

### CODE REVIEW

File: charts/internal/rsyslog-relp-configurator/templates/configmap.yaml [hunk 1]
```diff
@@ -63,6 +63,168 @@ data:
...
+          ;;
+      esac
+      shift
+    done
+
+    process_json() {
+      local json=$1
+
+      echo $json | \
+        jq -r '
```
[the remark this answers is not in the document]
Comment (plkokanov) [reply charts/internal/rsyslog-relp-configurator/templates/configmap.yaml:90]: Yes, we already depend on it in other scripts: https://github.com/gardener/gardener/blob/47570e118ff3dc746f4accfcf5e7649f4c7ba0be/pkg/component/extensions/operatingsystemconfig/original/components/kubelet/templates/scripts/health-monitor.tpl.sh#L60

File: pkg/controller/lifecycle/monitoring.go [hunk 2]
```diff
@@ -0,0 +1,2638 @@
...
+func deployMonitoringConfig(ctx context.Context, client client.Client, namespace string) error {
+	monitoring := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-prometheus", constants.ServiceName),
+			Namespace: namespace,
+		},
+		Data: map[string]string{
+			v1beta1constants.PrometheusConfigMapScrapeConfig:   monitoringScrapeConfig,
+			v1beta1constants.PrometheusConfigMapAlertingRules:  fmt.Sprintf("rsyslog-relp.rules.yaml: |\n  %s\n", utils.Indent(monitoringAlertingRules, 2)),
+			v1beta1constants.PlutonoConfigMapOperatorDashboard: fmt.Sprintf("rsyslog-relp-dashboard.json: '%s'", dashboardConfig),
```
[the remark this answers is not in the document]
Comment (plkokanov) [reply pkg/controller/lifecycle/monitoring.go:2624]: That's not necessary in this case. The created `shoot-rsyslog-relp-prometheus` `ConfigMap` looks like this:
```
data:
  dashboard_operators: |-
    rsyslog-relp-dashboard.json: '{
      "annotations": {
        "list": [
          {
            "builtIn": 1,
            "datasource": "-- Plutono --",
```

This `ConfigMap` is read and added to the `plutono-dashboards` `ConfigMap` which is loaded into plutno where the json data is on na single line and looks like this:
```
data:
  rsyslog-relp-dashboard.json: '{"annotations":{"list":[{"builtIn":1,"datasource":" ...
```

[the remark this answers is not in the document]
Comment (acumino) [reply pkg/controller/lifecycle/monitoring.go:2624]: Ok 👍 .

File: charts/internal/rsyslog-relp-configurator/templates/configmap.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (acumino) [reply charts/internal/rsyslog-relp-configurator/templates/configmap.yaml:90]: 👍

**Review by acumino — COMMENTED** [review]
/lgtm

File: pkg/controller/lifecycle/monitoring.go [hunk 3]
```diff
@@ -0,0 +1,2638 @@
...
+    target_label: __metrics_path__
+    replacement: /api/v1/namespaces/kube-system/pods/${1}:${2}/proxy/metrics
+  metric_relabel_configs:
+  - source_labels: [ __name__ ]
+    action: keep
+    regex: rsyslog_.+
+`
+
+	dashboardConfig = `{
+  "annotations": {
```
[the remark this answers is not in the document]
Comment (acumino) [reply pkg/controller/lifecycle/monitoring.go]: I can't resolve the comment please resolve it.

**Pull request merged**